### DISCUSSION TITLE
Add DHL Fastest Pitstop Award

### DISCUSSION THREAD

**Discussion by dlux80** [f1db/f1db#49] [Ideas]
First of all, great work being done here.

I have a question about a new feature, the DHL Fastest Pitstop Award has been around for a few years, is it possible to maintain the data for it?

Unfortunately, I only found the detailed data for 2023 and 2024 at https://inmotion.dhl/en/formula-1/fastest-pit-stop-award, I have the top 10 up to the introduction of the points system (2017), but without the lap, just time, driver, team and points. For 2016, I only have the top 5 and for 2015, even just the fastest stop of each race.

If the data finds its way into the great project, it could either be added to the existing data on the pit stops - stop time, points, placement or maintained in a separate file. In addition, an up-to-date status after each race would be desirable

**Comment by marceloverdijk** [comment]
Interesting idea @dlux80 , I will look into it myself as well.

**Comment by dlux80** [comment]
Hello,

for me it works to write the data into a database table and create a view. While I was checking the data, I noticed a discrepancy between the DHL data and the pit stop data (race_id 1099, driver_id 'oscar-piastri', lap 23 had to be corrected to 24). I also noticed that there is no pit stop data for the race with race_id 1115 (Belgium GP 2024) - I will open an issue for this. I am attaching two files to the post, the pure DHL Pitstop Award data for 2023 and 2024 and an export from my customized database, which contains the entries for the pitstop.yml with additions (stoptime, points, ranking) as well as the race_id for assignment.
[view_pitstops_with_awards_2023-2024.csv](https://github.com/user-attachments/files/18309880/view_pitstops_with_awards_2023-2024.csv)
[dhl_pitstop_award_2023-2024.csv](https://github.com/user-attachments/files/18309881/dhl_pitstop_award_2023-2024.csv)


--
Edit:
After the fixed data set:
[completed_view_pitstops_with_award_2023-2024.csv](https://github.com/user-attachments/files/18312962/completed_view_pitstops_with_award_2023-2024.csv)

**Comment by dlux80** [comment]
Here are added versions of 'pitstop.yml' and 'pitstopaward_standings.yml'. I have never created YAML files before, but maybe they will help. If adjustments are needed, I am open to making them.
[f1db_pitstopaward.zip](https://github.com/user-attachments/files/18341319/f1db_pitstopaward.zip)

**Reply by marceloverdijk** [reply]
Thx @dlux80 ,
For your information, I'm trying to reach out to DHL InMotion (without any luck), to see if we can get the raw data for all the years.
Another aspect to think about is, how quick do they provide this data after a race?

**Reply by dlux80** [reply]
I also made a request more than a week ago and have not received a response yet.

I retrieved the last data on December 11th, but I did not check before.

**Comment by dlux80** [comment]
For anyone who might need it, here is the data for the Pit-Stop-Award 2023 to 2025. 
[pit_stop_award.csv](https://github.com/user-attachments/files/24023198/pit_stop_award.csv)
[yaml.zip](https://github.com/user-attachments/files/24023202/yaml.zip)

PS.: The data was online yesterday before the f1db update.

**Comment by dlux80** [comment]
I used the source https://github.com/TracingInsights-Archive/PitStops to integrate the data for the years 2018–2022. I spot-checked the source against the API, and once the data matched, I combined it with F1DB. The result is included in the SQLite version and available as a CSV; YAML files will follow shortly.

[pit_stop_award.csv](https://github.com/user-attachments/files/29100149/pit_stop_award.csv)
[f1db_extra.db.zip](https://github.com/user-attachments/files/29100172/f1db_extra.db.zip)

And the missing YAML files:
[yml_pit_stop_award.zip](https://github.com/user-attachments/files/29132048/yml_pit_stop_award.zip)
